packages/backend/src/build-disk-image.ts

```diff
@@ -211,33 +217,68 @@ export async function buildDiskImage(imageData: unknown, history: History) {
         // ###########
         // Regardless what happens, we will need to clean up what we started (if anything)
         // which could be containers, volumes, images, etc.
-        await containerUtils.removeContainerAndVolumes(image.engineId, buildImageContainer.name);
+        if (buildImageContainer.name) {
+          await containerUtils.removeContainerAndVolumes(build.engineId, buildImageContainer.name);
+        }
       }
 
       // Mark the task as completed
       progress.report({ increment: -1 });
       telemetryLogger.logUsage('buildDiskImage', telemetryData);
 
       if (successful) {
+        try {
+          // Update the image build status to success
+          build.status = 'success';
+          await history.addOrUpdateBuildInfo(build);
+        } catch (e) {
+          // If for any reason there is an error.. (example, unable to write to history file)
+          // we do not want to stop the notification to the user, so
+          // just output this to console and continue.
+          console.error('Error updating image build status to success', e);
+        }
+
+        // Notify the user that the image has been built successfully
         await extensionApi.window.showInformationMessage(
           `Success! Your Bootable OS Container has been succesfully created to ${imagePath}`,
           'OK',
         );
       } else {
+        try {
+          // Update the image build status to error
+          build.status = 'error';
+          await history.addOrUpdateBuildInfo(build);
+        } catch (e) {
+          // Same as above, do not want to block other parts of the build
+          // so just output to console.
+          console.error('Error updating image build status to error', e);
```

Comment:
I think it would be interesting to know what build it is in the error

else from end user POV it's not really helpful when you read 'Error updating image build'